### DISCUSSION TITLE
Emits-Change flag on property

### DISCUSSION THREAD

**Discussion by fwilmsh** [Kistler-Group/sdbus-cpp#249] [Q&A] [answered]
Hi all!
I'm pretty new to sdbus-cpp and dbus in general. I need some direction on the following issue.
I'm currently hacking on the example code for the DBus server and managed to add and read/write a property to one of its interfaces. 
The property also gets assigned the emits-changed flag upon creation. 
But whenever I write something new to the property I can't seem to detect any signal on the bus when using busctl monitoring. 
Do I need to add something to my code in order for the server to emit the signal whenever the property gets changed?
Or is it something which should be handled automatically somehow by sbus-cpp or even libsdbus. 

Thanks in advance
Florian

**Comment by Oxymoron79** [comment, marked as answer]
There is a small section about the usage of DBus Standard Interfaces in the documentation: https://github.com/Kistler-Group/sdbus-cpp/blob/master/docs/using-sdbus-c%2B%2B.md#standard-d-bus-interfaces

The short summary for properties interface is: In your adaptor implementation, you need to add the `sdbus::Properties_adaptor` to the template parameters of `sdbus::AdaptorInterfaces`. This adds the `emitPropertiesChangedSignal` method to your adaptor that you can call to emit the signal on the `org.freedesktop.DBus.Properties` interface.

I hope that helps.

**Reply by fwilmsh** [reply]
I actually did use my own adaptor class for properties which is given an IObject reference, but the generated adapter code was still the right clue, so thank you @Oxymoron79 ;)
